mlir/lib/Util/Util.cpp

```diff
@@ -522,4 +521,62 @@ Operation *air::getAffineIfNestAndSpatialLoopFromOp(
   spatial_loop = parent;
   parent = parent->getParentOp();
   return parent;
+}
+
+// Check if an operand of an operation is read or write access
+char air::checkOpOperandReadOrWrite(Value v, Operation * owner) {
+  for (auto &op_operand : owner->getOpOperands()){
+    if (op_operand.is(v)){
+      return checkOpOperandReadOrWrite(op_operand);
+    }
+  }
+  // Value is not an opoperand of the operation
+  return 'e';
+}
+char air::checkOpOperandReadOrWrite(mlir::OpOperand &op_operand) {
+  auto owner = op_operand.getOwner();
+  // If used in DmaMemcpy Op
+  if (auto dma = dyn_cast<xilinx::air::DmaMemcpyInterface>(owner)) {
+    if (op_operand.is(dma.getSrcMemref())) {
+      return 'r';
+    } else if (op_operand.is(dma.getDstMemref())) {
+      return 'w';
+    } else {
+      assert(false && "Unknown operand in air.dma");
```

Comment:
This is not specific to this PR as it appears all over the place.
But I am curious about what is the expected plan for this when not compiling with `assert` on? To go on? Because there is no test in the `assert` so it it is obvious it will goes badly later. :-)
Would it be clearer to have a simple `std::terminate`? Or throwing an exception just to capture the error message even if it is compiled with `-fnoexcept` (I am unsure whether it can work).

Reply:
I agree with you. Currently I used assertion here in the draft for debugging, but I will later review the codebase and use "emitError" MLIR diagnostic to replace those assertions before I push. Thanks for your comment.